packages/react-google-maps-api/src/LoadScriptNext.tsx

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react'
+
+import { LoadScriptProps, DefaultLoadingElement } from './LoadScript'
+import { useLoadScript } from './useLoadScript'
+
+export interface LoadScriptNextProps extends LoadScriptProps {
+  loadingElement: React.ReactElement
+  children: React.ReactElement
+}
+
+export const LoadScriptNext = React.memo<LoadScriptNextProps>(
+  ({ loadingElement = <DefaultLoadingElement />, children, ...props }) => {
```

Comment:
Please don't use default values for props on destructuring and use defaultProps instead. performance requirements.

Reply:
How do you handle that with TypeScript? I mean if I do `function LoadScriptNext` it does not have `defaultProps` and I would have to `ts-ignore` it which is ugly. The only way I know is to keep `const` which will keep the name as well, but allow proper typing.

```tsx
const LoadScriptNext: React.FC<LoadScriptNextProps> = ({ loadingElement, children, ...props }) => {
  const { isLoaded } = useLoadScript(props)
  return isLoaded ? children : loadingElement
}

LoadScriptNext.defaultProps = {
  loadingElement: <DefaultLoadingElement />
}

export default React.memo(LoadScriptNext)

```

Reply:
Another approach would be like that, but I think const typing is much easier and approachable.

```ts
const defaultLoadingElement = <DefaultLoadingElement />

function LoadScriptNext({ loadingElement, children, ...props }: LoadScriptNextProps) {
  const { isLoaded } = useLoadScript(props)
  return isLoaded ? children : loadingElement || defaultLoadingElement
}

export default React.memo(LoadScriptNext)
```

Reply:
```jsx
function LoadScriptNext(props: LoadScriptNextProps) {
  const { isLoaded } = useLoadScript(props)

  return isLoaded ? props.children : props.loadingElement
}

LoadScriptNext.defaultProps = {
  loadingElement: <DefaultLoadingElement />
}

export default React.memo(LoadScriptNext)
```

Reply:
And what's the benefit of using `defaultProps` really? It did make sense for class components, but here it feels rather obsolete.

Reply:
defaultProps does not get redeclared in compare with destructuring default values

Reply:
basic types is pretty much safe, but objects, arrays and functions as default values recreated on each function call :-/

Reply:
Sure, if you do `const defaultLoadingElement = <DefaultLoadingElement />` and referenced that, it's essentially the same thing, just bit more manual.

Reply:
Component.defaultProps is not deprecated, and it is a recommended way to set default props :-)

Reply:
Recommended? Would love to see docs link explaining that? :) 

![image](https://user-images.githubusercontent.com/1096340/59044583-b1b1ff00-887e-11e9-93d6-fff7d6246e2e.png)